File: DotnetAuth/Data/ResponseModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DotnetAuth.Data
{
    public class ResponseModel
    {
        public ResponseModel(ResponseCode code, string message, object dataSet)
        {
            Code = code;
            Message = message;
            DataSet = dataSet;
        }
        public ResponseCode Code { get; set; }
        public string Message { get; set; }
        public object DataSet { get; set; }
    }
}

File: DotnetAuth/client/src/app/_models/response-model.ts
import { ResponseCode } from "./enums";

export class ResponseModel {
    code: ResponseCode = ResponseCode.NotSet;
    message: string = '';
    dataSet: any;

    // constructor(code: ResponseCode, message: string) {
    //     this.message = message;
    //     this.code = code;
    // }
}